crates/compute/src/layer.rs

```diff
@@ -599,6 +599,10 @@ pub type FSlice<'a, F, HAL> = <<HAL as ComputeLayer<F>>::DevMem as ComputeMemory
 pub type FSliceMut<'a, F, HAL> =
 	<<HAL as ComputeLayer<F>>::DevMem as ComputeMemory<F>>::FSliceMut<'a>;
 
+pub type KernelMem<F, HAL> = <<HAL as ComputeLayer<F>>::KernelExec as KernelExecutor<F>>::Mem;
```

Comment:
Do any of the HAL's kernel-specific functions need to be updated to accept these new types as inputs?